packages/kbn-plugin-generator/sao_template/template/public/plugin.ts

```diff
@@ -0,0 +1,42 @@
+import { i18n } from '@kbn/i18n';
+import { AppMountParameters, CoreSetup, CoreStart, Plugin } from '<%= relRoot %>/src/core/public';
+import { <%= upperCamelCaseName %>PluginSetup, <%= upperCamelCaseName %>PluginStart, AppPluginDependencies } from './types';
+import { PLUGIN_NAME } from '../common';
+
+export class <%= upperCamelCaseName %>Plugin
+  implements Plugin<<%= upperCamelCaseName %>PluginSetup, <%= upperCamelCaseName %>PluginStart> {
+  
+  public setup(core: CoreSetup): <%= upperCamelCaseName %>PluginSetup {
+    // Register an application into the side navigation menu
+    core.application.register({
+      id: '<%= camelCase(name) %>',
+      title: PLUGIN_NAME,
+      async mount(params: AppMountParameters) {
+        // Load application bundle
+        const { renderApp } = await import('./application');
+        // Get start services as specified in kibana.json
+        const [coreStart, depsStart] = await core.getStartServices();
+        // Render the application
+        return renderApp(coreStart, depsStart as AppPluginDependencies, params);
```

Comment:
You can avoid the manual casting by specifying the plugin's dependencies:

`Plugin<<%= upperCamelCaseName %>PluginSetup, <%= upperCamelCaseName %>PluginStart>`

to 

`Plugin<<%= upperCamelCaseName %>PluginSetup, <%= upperCamelCaseName %>PluginStart>, {}, AppPluginDependencies `

Reply:
Unfortunately, I think you also have to specify the generic on `CoreSetup` on line 9, because TS does not infer types for function arguments on interface implementations.

Line 9 could be:
```
public setup(core: CoreSetup<AppPluginDependencies>): <%= upperCamelCaseName %>PluginSetup {
```